### PR TITLE
2.3 dev nuget

### PR DESCRIPTION
**Pull request by nurmi**
adds references to dotnet nuget in areas where non-os package content and vulnerability queries and capabilities are discussed.

### CODE REVIEW

File: content/docs/engine/engine_installation/storage/layer_caching.md [hunk 1]
```diff
@@ -17,7 +17,7 @@ As part of image analysis the Anchore Engine will:
     - Digest of every file (SHA1, SHA256 and MD5)
     - File attributes (size, owner, permissions, etc)
     - Operating System package manifest
-    - Software library package manifest  (NPM, GEM, Java, Python)
+    - Software library package manifest  (NPM, GEM, Java, Python, Nuget)
```
Comment (alfredodeza) [inline content/docs/engine/engine_installation/storage/layer_caching.md:20]: Composer too, no?

File: content/docs/overview/concepts/_index.md [hunk 2]
```diff
@@ -17,7 +17,7 @@ weight: 2
 
 - On-Premises UI for visualization, policy editing, report generation, and user management
 - RBAC support for the APIs
-- Proprietary vulnerability data feeds to significantly enhance vulnerability detection in application packages and libraries (e.g. python, ruby gems, nodejs packages, java jars...)
+- Proprietary vulnerability data feeds to significantly enhance vulnerability detection in application packages and libraries (e.g. python, ruby gems, nodejs packages, java jars, dotnet nuget packages...)
```
Comment (alfredodeza) [inline content/docs/overview/concepts/_index.md:20]: Can we normalize on either _dotnet_ or _Dotnet_ ?

File: content/docs/engine/engine_installation/storage/layer_caching.md (hunk 1 above)
Comment (zhill) [reply content/docs/engine/engine_installation/storage/layer_caching.md:20]: Anchore doesn't analyze composer packages yet

File: content/docs/overview/concepts/_index.md (hunk 2 above)
Comment (zhill) [reply content/docs/overview/concepts/_index.md:20]: I think we can just use NuGet (that is their capitalization). .NET is implied. But if referenced it generally is .NET, not dotnet. I think that is used in command-line and elsewhere you cannot use a period, but the framework is specifically '.NET'

**Comment by alfredodeza** [comment]
> Across the board let's use "NuGet" as the name, and C#/.NET as the language as I think those are probably most correct

Do you mean literally "C#/.NET" or should I read that as "use C# or .NET" ?

**Pull request merged**